### ISSUE TITLE
Picking a date column for sorting results in "day" granularity sort order

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#2531]
When selecting a date column in the sort picker we incorrectly include "day" as the unit, causing rows to be sorted by day but sometimes not sorted correctly within each day.

Note that clicking a table header doesn't cause the unit to be included, so it sorts correctly. This can be used as a workaround until we fix this issue.

**Comment by agilliland** [comment]
@tlrobinson can you explain a bit more how you are reproducing this?  I'm not seeing this happen.

**Comment by tlrobinson** [comment]
Incorrect, when using the sort picker:

![screenshot 2016-05-06 10 36 44](https://cloud.githubusercontent.com/assets/18193/15081240/d75644ee-1376-11e6-9954-bfdcc8d84502.png)

Correct, when clicking a column header:

![screenshot 2016-05-06 10 39 04](https://cloud.githubusercontent.com/assets/18193/15081241/d98bc91e-1376-11e6-8487-52c2a62d71c0.png)

**Comment by agilliland** [comment]
So, this is ultimately an issue on the backend where running a `:rows` query with a simple order-by clause on a datetime field is incorrectly forcing that field representation to be grouped by day.

https://github.com/metabase/metabase/blame/master/src/metabase/query_processor/resolve.clj#L106

For example when I run the above query I see the following, which is not the correct SQL because the order by clause should not be casting the created_at column to a date:

```
SELECT "PUBLIC"."ORDERS"."USER_ID" AS "USER_ID", "PUBLIC"."ORDERS"."TOTAL" AS "TOTAL", "PUBLIC"."ORDERS"."TAX" AS "TAX", "PUBLIC"."ORDERS"."SUBTOTAL" AS "SUBTOTAL", "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", "PUBLIC"."ORDERS"."ID" AS "ID", "PUBLIC"."ORDERS"."CREATED_AT" AS "CREATED_AT"
FROM "PUBLIC"."ORDERS"
ORDER BY CAST("PUBLIC"."ORDERS"."CREATED_AT" AS DATE) ASC
LIMIT 2000
```

**Comment by tlrobinson** [comment]
@agilliland So I think we agreed the correct fix is:

1. ensure the frontend uses a bare `field-id` rather than `datetime_field` with a granularity
2. ensure the backend doesn't group by day when sorting by a datetime column without a `datetime_field` and granularity 

I can handle 1, do you or @camsaul want to take 2?

**Comment by salsakran** [comment]
@camsaul raw rows should definitely not cast the timestamp to a date. This is incorrect, and should be fixed.

We can and should default to timezone when charting aggregates by a timestamp, but in this case it is incorrect.

**Comment by camsaul** [comment]
Ok @tlrobinson part 2 is merged so I'm going to assign this back to you so you can tackle part 1.

**Comment by salsakran** [comment]
Re-opening as the fix for this wasn't correct. 

The resulting behavior is incorrect and a hard blocker on 0.17.1

**Comment by camsaul** [comment]
Fixed by #2586
